base/string.jl

```diff
@@ -968,8 +968,8 @@ byte_string_classify(s::ByteString) = byte_string_classify(s.data)
     # 1: valid ASCII
     # 2: valid UTF-8
 
-is_valid_ascii(s::Union(Array{UInt8,1},ByteString)) = byte_string_classify(s) == 1
-is_valid_utf8(s::Union(Array{UInt8,1},ByteString)) = byte_string_classify(s) != 0
+isvalid(::Type{ASCIIString}, s::Union(Array{UInt8,1},ByteString)) = byte_string_classify(s) == 1
```

Comment:
> About adding more specialized methods, is there any way of telling just where the old code might still be getting used? There is so much in Base, that it seems like quite a lot might already be going directly to the old code, no matter what I do, not just some Win32 calls.

Maybe just deprecating them? That should print a backtrace for every place the function is called from. Not sure about the interaction with precompilation, though, so you may want (re)move the sysimage before starting Julia.